Adapt obsolete `adb` instructions

karlhorky opened the pull request:
# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.
-->

We have had multiple students report `adb` fails for them on the command line, but Expo still works just fine with their virtual devices

# How

<!--
How did you build this feature or fix this bug and why?
-->

Tested with multiple students

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

Tested with multiple students

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

karlhorky committed:
Remove obsolete instructions

EvanBacon commented on docs/pages/workflow/android-studio-emulator.md:
````diff
@@ -39,8 +39,6 @@ echo "export PATH=$HOME/Library/Android/sdk/platform-tools:\$PATH" >> ~/`[[ $SHE
 source ~/`[[ $SHELL == *"zsh" ]] && echo '.zshenv' || echo '.bash_profile'`
 ```
 
-- Finally, make sure that you can run `adb` from your terminal.
````
The reason this works is because we historically shipped adb with expo-cli but this has caused other issues regarding versioning so we may pull out the vendored copy of adb.

I imagine the issue is that some users have adb installed differently when Android Studio is installed. We could update the doc to run `$ANDROID_HOME/platform-tools/adb || adb` which would check both locations.

karlhorky replied:
Oh, interesting, to be clear about the error they are receiving, it has to do with a segmentation fault (error message: `Segmentation fault 11`)

![image](https://user-images.githubusercontent.com/1935696/156663165-840fdc7a-a6c8-49e6-8f74-56bda72e9103.png)

karlhorky replied:
With `$ANDROID_HOME` you mean `$ANDROID_SDK`, right?

karlhorky replied:
They both fail with `Segmentation fault: 11` for these students:

![image](https://user-images.githubusercontent.com/1935696/156762924-8e63c6c5-3825-4908-9b4a-053900a7891b.png)

karlhorky replied:
To be clear, these are the same students that were able to start the virtual device and get Expo apps running on it.

karlhorky replied:
@EvanBacon thoughts on this? I'm happy to do a different PR if necessary, but I think that if both of them are failing, then it doesn't make sense to recommend running this command...

karlhorky replied:
Just circling back around to this - @EvanBacon would you offer your thoughts here?

karlhorky committed:
Avoid segmentation fault

karlhorky commented:
@EvanBacon after some testing, I found that running `adb --version` does not cause the segmentation fault (see output below), so I've added back the step and added this flag. What do you think?

```bash
$ adb --version
Android Debug Bridge version 1.0.41
Version 32.0.0-8006631
Installed as /Users/k//Library/Android/sdk/platform-tools/adb
```
